### DISCUSSION TITLE
Bind with opposite value

### DISCUSSION THREAD

**Discussion by laserir** [zauberzeug/nicegui#1621] [Q&A] [answered]
### Question

Dear all,
do I need a custom class for this "logic" or can it be done otherwise?:

I have a ui.switch() which is bound to multiple objects:

```py
login_switch = ui.switch()
with ui.row().bind_visibility_from(login_switch, 'value'):
            ui.icon('sensors')
with ui.row().bind_visibility_from(login_switch, 'value'):
            ui.button('Login', on_click=login_function)
```

I want the first row (with the icon) to be visible if the switch is DISABLED, the second row behaves like I want (switch enabled -> button visible).

Any ideas?

**Comment by natankeddem** [comment]
This is a bit hacky, but maybe it works for you?
```py
from nicegui import ui

login_switch = ui.switch(on_change=lambda e: not_login_switch.set_value(not e.value))
not_login_switch = ui.switch(value=not login_switch.value)
not_login_switch.visible=False
with ui.row().bind_visibility_from(login_switch, 'value'):
            ui.label('bye')
with ui.row().bind_visibility_from(not_login_switch, 'value'):
            ui.label('hi')

ui.run()
```

**Comment by laserir** [comment]
Works perfectly ;) Thanks a lot!

**Comment by falkoschindler** [comment, marked as answer]
NiceGUI has a non-hacky, built-in solution for your problem.

You can specify a transformation function when binding values (in this case called `backward` because it's the "backward" direction):
```py
with ui.row().bind_visibility_from(login_switch, 'value', backward=lambda value: not value):
    ui.icon('sensors')
```
For binding visibility there's even a shortcut. You can specify the `value` for which the element should be visible:
```py
with ui.row().bind_visibility_from(login_switch, 'value', value=False):
    ui.icon('sensors')
```

**Comment by laserir** [comment]
@falkoschindler This is perfect! Just what I searched for! :)) Thanks!!
